modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js

```diff
@@ -0,0 +1,249 @@
+import ClayButton from '../shared/ClayButton.es';
+import ClaySpinner from '../shared/ClaySpinner.es';
+import Conjunction from './Conjunction.es';
+import CriteriaBuilder from './CriteriaBuilder.es';
+import CriteriaSidebar from '../criteria_sidebar/CriteriaSidebar.es';
+import EmptyPlaceholder from './EmptyPlaceholder.es';
+import getCN from 'classnames';
+import HTML5Backend from 'react-dnd-html5-backend';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+	conjunctionShape,
+	contributorShape,
+	operatorShape,
+	propertyGroupShape,
+	propertyTypesShape
+} from '../../utils/types.es';
+import {DragDropContext as dragDropContext} from 'react-dnd';
+import {getPluralMessage} from '../../utils/utils.es';
+
+class ContributorBuilder extends React.Component {
+	static propTypes = {
+		contributors: PropTypes.arrayOf(contributorShape),
+		editing: PropTypes.bool.isRequired,
+		emptyContributors: PropTypes.bool.isRequired,
+		membersCount: PropTypes.number,
+		membersCountLoading: PropTypes.bool,
+		onConjunctionChange: PropTypes.func,
+		onPreviewMembers: PropTypes.func,
+		onQueryChange: PropTypes.func,
+		previewMembersURL: PropTypes.string,
+		propertyGroups: PropTypes.arrayOf(propertyGroupShape),
+		supportedConjunctions: PropTypes.arrayOf(conjunctionShape).isRequired,
+		supportedOperators: PropTypes.arrayOf(operatorShape).isRequired,
+		supportedPropertyTypes: propertyTypesShape.isRequired
+	};
+
+	static defaultProps = {
+		contributors: [],
+		membersCount: 0,
+		onConjunctionChange: () => {},
+		onPreviewMembers: () => {},
+		onQueryChange: () => {},
+		membersCountLoading: false
+	};
+
+	constructor(props) {
+		super(props);
+
+		const {contributors, propertyGroups} = props;
+
+		const firstContributorNotEmpty = contributors.find(
+			contributor => contributor.query !== ''
+		);
+
+		const propertyKey = firstContributorNotEmpty
+			? firstContributorNotEmpty.propertyKey
+			: propertyGroups[0].propertyKey;
+
+		this.state = {
+			editingId: propertyKey
+		};
+	}
+
+	_handleCriteriaChange = (criteriaChange, index) => {
+		const {onQueryChange} = this.props;
+		onQueryChange(criteriaChange, index);
```

Comment:
We should have a new line between above this.

General rule of thumb for newlines:
- If order matters, add a new line
- If ordering doesn't matter, no new line.

For Example:
```js
const {onQueryChange} = this.props;
onQueryChange(criteriaChange, index);
// Ordering matters here ^ because if you switched the lines, it would break
onQueryChange(criteriaChange, index);
const {onQueryChange} = this.props;
```
One big reason for this, is because sometimes the source formatter or sometimes Brian will automatically sort things that have no line break, and it'll cause issues.

Lets try to apply this consistently throughout.